components/proxy/src/main/java/com/hotels/styx/infrastructure/YamlReader.java

```diff
@@ -1,151 +0,0 @@
-
-  Unless required by applicable law or agreed to in writing, software
-  distributed under the License is distributed on an "AS IS" BASIS,
-  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-  See the License for the specific language governing permissions and
-  limitations under the License.
- */
-package com.hotels.styx.infrastructure;
-
-import com.fasterxml.jackson.core.type.TypeReference;
-import com.fasterxml.jackson.databind.JsonNode;
-import com.fasterxml.jackson.databind.ObjectMapper;
-import com.fasterxml.jackson.databind.node.ObjectNode;
-import com.fasterxml.jackson.dataformat.yaml.YAMLFactory;
-import com.hotels.styx.api.Resource;
-import com.hotels.styx.infrastructure.configuration.UnresolvedPlaceholder;
-import com.hotels.styx.infrastructure.configuration.yaml.NodePath;
-import com.hotels.styx.infrastructure.configuration.yaml.PlaceholderResolver;
-
-import java.io.IOException;
-import java.util.Collection;
-import java.util.List;
-import java.util.Map;
-import java.util.Optional;
-
-import static com.fasterxml.jackson.core.JsonParser.Feature.AUTO_CLOSE_SOURCE;
-import static com.fasterxml.jackson.databind.DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES;
-import static com.google.common.base.Preconditions.checkNotNull;
-import static com.google.common.base.Preconditions.checkState;
-import static com.google.common.base.Throwables.propagate;
-import static com.google.common.io.ByteStreams.toByteArray;
-import static com.hotels.styx.api.io.ResourceFactory.newResource;
-import static com.hotels.styx.infrastructure.configuration.yaml.PlaceholderResolver.extractPlaceholders;
-import static com.hotels.styx.infrastructure.configuration.yaml.PlaceholderResolver.replacePlaceholder;
-import static com.hotels.styx.infrastructure.configuration.yaml.PlaceholderResolver.resolvePlaceholders;
-import static java.util.Collections.emptyList;
-import static java.util.Collections.emptyMap;
-import static java.util.stream.Collectors.toList;
-import static java.util.stream.StreamSupport.stream;
-
-/**
- * Reads a YAML file and convert it to specified class.
- * It can handle included files by collecting and adding them to the original set if the class implements
- * {@link com.hotels.styx.infrastructure.Composite} interface.
- *
- * @param <T>
- */
-public final class YamlReader<T> {
-    private static final ObjectMapper MAPPER = new ObjectMapper(new YAMLFactory())
-            .disable(FAIL_ON_UNKNOWN_PROPERTIES)
-            .configure(AUTO_CLOSE_SOURCE, true);
-
-    private final Map<String, String> overrides;
-
-    public YamlReader() {
-        this(emptyMap());
-    }
-
-    public YamlReader(Map<String, String> overrides) {
-        this.overrides = checkNotNull(overrides);
-    }
-
-    public T read(byte[] content, TypeReference<T> typeReference) throws Exception {
-        JsonNode rootNode = MAPPER.readTree(content);
-        List<String> includePaths = includedPaths(rootNode);
```

Comment:
Do you know this `includePaths` handing does? I don't see that having been moved down tho the call site.

Reply:
In practice, it does nothing. The origins file is supposed to be a YAML representation of a `List` of `BackendService`s. If an `include` property were added, it would change the structure and no longer be a valid list.